### PR TITLE
Dev/support fsdp2

### PR DESCRIPTION
**Pull request by haiasd**
1. fix vllm v0 bug
2. fix reward replica bug
3. support fsdp2
4. support token-wise pg loss
5. temporary manual grad clip
6. bucket-wise parameter synchronization

### CODE REVIEW

**Review by lostkevin — APPROVED** [review]

**Review by yytang0204 — APPROVED** [review]

File: chatlearn/algorithm/grpo_utils/policy_trainer.py [hunk 1]
```diff
@@ -225,21 +239,28 @@ def train_step(self, data_list):
...
-            .detach()
-            .item()
-        )
+            # compute backward loss
+            pg_loss_mean = torch.sum(pg_loss) / response_token_length_total * self.fsdp_size
+            if self.module_args.entropy_coef > 0:
+                pg_loss_mean = pg_loss_mean - self.module_args.entropy_coef * entropy_loss_mean
+            if self.module_args.kl_coef > 0:
+                pg_loss_mean = pg_loss_mean + self.module_args.kl_coef * kl_loss_mean
+            pg_loss_mean.backward()
```
Comment (lostkevin) [inline chatlearn/algorithm/grpo_utils/policy_trainer.py:251]: total_loss

**Review by yytang0204 — APPROVED** [review]

**Review by lostkevin — APPROVED** [review]
